src/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -634,14 +634,34 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_G()
             Assert.Equal(expectedString, result.ToString("g"));
         }
 
-        [Fact]
-        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
+        [Theory]
+        [MemberData("Format_String_TestData_O")]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O(DateTime dt, string expected)
         {
-            DateTime expected = DateTime.MaxValue;
-            string expectedString = expected.ToString("o");
+            DateTime result = DateTime.ParseExact(dt.ToString("o"), "o", null, DateTimeStyles.None);
+            Assert.Equal(expected, result.ToString("o"));
+        }
 
-            DateTime result = DateTime.ParseExact(expectedString, "o", null, DateTimeStyles.None);
-            Assert.Equal(expectedString, result.ToString("o"));
+        public static IEnumerable<object[]> Format_String_TestData_O()
```

Comment:
I think this whole set of tests could be optimized around better sets of data and sharing of things between tests. Let track that work here: https://github.com/dotnet/corefx/issues/13170